File: LeetCode/Medium/0945-minimum-increment-to-make-array-unique/0945-minimum-increment-to-make-array-unique.cpp
class Solution {
public:
    int minIncrementForUnique(vector<int>& nums) {
        int ret = 0;
        vector<int> count(100001);

        for (int v : nums) { ++count[v]; }

        for (int i = 0; i < 100000; i++) {
            if (count[i] > 1) {
                int temp = count[i] - 1;
                ret += temp;
                count[i+1] += temp;
            }
        }

        if (count[100000] > 1) {
            int temp = count[100000] - 1;
            ret += temp * (temp + 1) / 2;
        }

        return ret;
    }
};

File: LeetCode/Medium/0945-minimum-increment-to-make-array-unique/0945-minimum-increment-to-make-array-unique.py
class Solution:
    def minIncrementForUnique(self, nums: List[int]) -> int:
        ret = 0
        count = [0] * 100001

        for v in nums: count[v] += 1

        for i in range(100000):
            if count[i] > 1:
                temp = count[i] - 1
                ret += temp
                count[i+1] += temp

        if count[100000] > 1:
            temp = count[100000] - 1
            ret += temp * (temp + 1) // 2
        
        return ret